protocol/src/main/java/org/jboss/as/protocol/ProtocolConnectionManager.java

```diff
@@ -116,7 +116,9 @@ public void shutdown() {
                 connectTask.shutdown();
             }
         }
-        StreamUtils.safeClose(connection);
+        if (connection != null) {
+            connection.closeAsync();
+        }
```

Comment:
Note to self: this is probably slightly different as it doesn't suppress exceptions thrown by closeAsync.